packages/vitest/src/utils/coverage.ts

```diff
@@ -537,20 +537,41 @@ export class BaseCoverageProvider<Options extends ResolvedCoverageOptions<'istan
     const servers = [
       ...ctx.projects.map(project => ({
         root: project.config.root,
+        browser: project.browser?.vite,
         vitenode: project.vitenode,
       })),
       // Check core last as it will match all files anyway
-      { root: ctx.config.root, vitenode: ctx.vitenode },
+      { root: ctx.config.root, vitenode: ctx.vitenode, browser: ctx.getRootProject().browser?.vite },
     ]
 
     return async function transformFile(filename: string): Promise<TransformResult | null | undefined> {
       let lastError
 
-      for (const { root, vitenode } of servers) {
+      for (const { root, vitenode, browser } of servers) {
         if (!filename.startsWith(root)) {
           continue
         }
 
+        if (browser) {
+          const noDiscovery = browser.config.optimizeDeps.noDiscovery
+          const include = browser.config.optimizeDeps.include
+          browser.config.optimizeDeps.noDiscovery = true
+          browser.config.optimizeDeps.include = undefined
+
+          // Deprecated, should be equal to { noDiscovery, include } settings above
+          const disabled = browser.config.optimizeDeps.disabled
+          browser.config.optimizeDeps.disabled = true
+
+          const result = await browser?.transformRequest(filename).catch(() => null)
+          browser.config.optimizeDeps.noDiscovery = noDiscovery
+          browser.config.optimizeDeps.include = include
+          browser.config.optimizeDeps.disabled = disabled
```

Comment:
This is quite hacky and doesn't seem to work correctly. Vite will still cause side-effects for the dependencies of these files:

> [vite] (client) ✨ optimized dependencies changed. reloading

Though I'm unable to reproduce this locally. And have no idea how to fix this.

Reply:
These were probably overwritten by parallel requests caused by `Promise.all()`. By not restoring original values works.